source/_integrations/numato.markdown

```diff
@@ -0,0 +1,244 @@
+---
+title: Numato USB GPIO Expander
+description: Instructions on how to integrate a Numato GPIO expander into Home Assistant.
+ha_category:
+  - DIY
+  - Binary Sensor
+  - Sensor
+  - Switch
+logo: numato.png
+ha_release: 0.108.0
+ha_iot_class: Local Push
+ha_domain: numato
+---
+
+The `numato` integration is the base for all related GPIO platforms of the
+[Numato 32 Port USB GPIO expander](https://numato.com/product/32-channel-usb-gpio-module-with-analog-inputs):
+
+* [Binary Sensor](#binary-sensor)
+* [Sensor](#sensor)
+* [Switch](#switch)
+
+No general setup is needed. Add the following line to your Home Assistant
+configuration to enable the integration:
+
+```yaml
+numato:
+```
+
+## Binary Sensor
+
+The `numato` binary_sensor platform allows you to operate the GPIOs of your
+[Numato](https://numato.com) 32 port USB GPIO expander in binary input mode.
+
+<div class='note warning'>
+As the Numato devices do not have internal pull-up or pull-down circuitry,
+be careful not to destroy a port by creating a short circuit. Refer to the
+<a href="https://numato.com/docs/32-channel-usb-gpio-module-with-analog-inputs/#gpio-with-switches-8">
+Numato documentation</a> on how to connect a switch to an input port for
+example.
+</div>
+
+### Configuration
+
+To use GPIO ports `2`, `3` and `4` of your [device with id](#device-ids) `0` as
+binary input ports, add the following to your `binary_sensor` configuration:
+
+```yaml
+binary_sensor:
```

Comment:
Only required config variables in the first example, that goes for any example below.

Reply:
Thanks for the hint. Is it some unwritten rule or can you point me to some docs (since I couldn't find it in the doc standards). Don't get me wrong, I'll change that. Just want to know the reasoning behind. Thanks for reviewing anyways!

Reply:
We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.